### PR TITLE
MINOR: [Java] Bump com.google.flatbuffers:flatbuffers-java from 23.5.26 to 24.3.25 in /java MANUAL

### PR DESCRIPTION
**Pull request by vibhatha**
### Rationale for this change

Flatbuffer upgrade requires regeneration of the code as suggested by dependabot [here](https://github.com/apache/arrow/pull/42185). 

### What changes are included in this PR?

Regenerates the code using the proposed upgrade.

### Are these changes tested?

Using existing tests. 

### Are there any user-facing changes?

No

### CODE REVIEW

**Comment by vibhatha** [comment]
@github-actions crossbow submit -g java

File: java/README.md [hunk 1]
```diff
@@ -72,6 +72,35 @@ find java/format/src -type f | while read file; do
 done
 ```
 
+# add module-info.java to the generated code
+
+Make sure to add the `java/format/src/main/java/module-format.java` as follows.
```
Comment (lidavidm) [inline java/README.md:77]: Er, wouldn't you just revert the deletion via Git?

Comment (vibhatha) [reply java/README.md:77]: Ahh 😌 we could sure.

Comment (vibhatha) [reply java/README.md:77]: I will remove that.

**Comment by vibhatha** [comment]
@lidavidm the failing crossbow is a reported issue https://github.com/apache/arrow/issues/42197

**Comment by vibhatha** [comment]
@lidavidm shall we re-run the failed crossbow?

**Comment by lidavidm** [comment]
@github-actions crossbow submit java-jars

**Comment by vibhatha** [comment]
@github-actions crossbow submit java-jars

**Comment by vibhatha** [comment]
@lidavidm now the CIs failing at https://github.com/ursacomputing/crossbow/actions/runs/9589770273/job/26444735376#step:6:39299

This bug is already reported: https://github.com/apache/arrow/issues/42208
@llama90 could we check what is happening here?

**Comment by llama90** [comment]
I will take a look!

**Comment by vibhatha** [comment]
Thank you @llama90

**Comment by lidavidm** [comment]
@vibhatha can we rebase here and validate the Crossbow tests?

**Comment by vibhatha** [comment]
Will do @lidavidm

**Comment by vibhatha** [comment]
@lidavidm rebased.

**Comment by vibhatha** [comment]
@github-actions crossbow submit -g java

**Comment by vibhatha** [comment]
@lidavidm CIs are passing.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**